rest/rest-sqs/src/main/scala/org/elasticmq/rest/stats/StatisticsDirectives.scala

```diff
@@ -0,0 +1,95 @@
+package org.elasticmq.rest.stats
+
+import java.util.concurrent.Executors
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes.NotFound
+import akka.http.scaladsl.server.Route
+import org.elasticmq.metrics.QueueMetricsOps
+import org.elasticmq.rest.sqs.QueueAttributesOps
+import org.elasticmq.rest.sqs.directives.ElasticMQDirectives
+import org.elasticmq.util.NowProvider
+import org.elasticmq.{QueueData, QueueStatistics}
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+final case class QueuesResponse(
+                                 name: String,
+                                 statistics: QueueStatisticsResponse
+                               )
+
+final case class QueueResponse(
+                                name: String,
+                                attributes: Attributes
+                              )
+
+final case class QueueStatisticsResponse(
+                                          approximateNumberOfVisibleMessages: Long,
+                                          approximateNumberOfInvisibleMessages: Long,
+                                          approximateNumberOfMessagesDelayed: Long
+                                        )
+
+
+trait JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val queueStatisticsFormat = jsonFormat3(QueueStatisticsResponse)
+  implicit val queuesFormat = jsonFormat2(QueuesResponse)
+  implicit val queueFormat = jsonFormat2(QueueResponse)
+}
+
+trait StatisticsDirectives extends JsonSupport {
+  this: ElasticMQDirectives with QueueAttributesOps  =>
+
+  lazy val ec: ExecutionContext = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))
+  implicit val duration = timeout.duration
+
+  def statistics(implicit np: NowProvider) = {
+    pathPrefix("statistics" / "queues") {
+      concat(
+        pathEndOrSingleSlash {
+          complete {
+            gatherAllQueuesWithStats
+          }
+        },
+        path(Segment) { queueName =>
+          gatherSpecificQueueWithAttributesRoute(queueName)
+        }
+      )
+    }
+  }
+
+  def gatherAllQueuesWithStats(implicit np: NowProvider): Future[Iterable[QueuesResponse]] = {
+
+    QueueMetricsOps.getQueuesStatistics(queueManagerActor, np)
+      .map { x => x.map { case (name, stats) => mapToRestQueuesResponse(name, stats) } }
+  }
+
+  def gatherSpecificQueueWithAttributesRoute(queueName: String): Route = {
+    val map = Map("QueueName" -> queueName)
+    queueActorAndDataFromRequest(map) { (queueActor, queueData) =>
```

Comment:
maybe this can be done simpler - looking  up the actor by name?

Reply:
I'll try it! Thanks!

Reply:
I've rechecked its usage. Function queueActorAndDataFromRequest gives also "queue data" that is used in the logic of gathering attributes. When I'll try to lookup for actors by name I will need to implement this exactly like the "queueActor..." method does. So I coming to the conclusion that maybe I should leave it that way like it is... ;) What do you think?

Reply:
OK got your point. I've rewritten it without using a map. Thanks!